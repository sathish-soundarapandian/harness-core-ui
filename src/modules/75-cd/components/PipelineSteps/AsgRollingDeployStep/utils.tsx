/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import * as Yup from 'yup'
import { getMultiTypeFromValue, MultiTypeInputType } from '@harness/uicore'
import type { UseStringsReturn } from 'framework/strings'

export const minimumHealthyPercentageSchema = (
  getString: UseStringsReturn['getString']
): Yup.Schema<number | undefined> => {
  return Yup.lazy((value: number): any => {
    /* istanbul ignore else */
    if (getMultiTypeFromValue(value as any) === MultiTypeInputType.FIXED) {
      return Yup.mixed().test({
        test(): boolean | Yup.ValidationError {
          if (!value) {
            return true
          }
          if (value > 100) {
            return this.createError({ message: getString('cd.minimumHealthyPercentageMaxLimit') })
          }
          if (value < 0) {
            return this.createError({ message: getString('cd.minimumHealthyPercentageMinLimit') })
          }
          return true
        }
      })
    }
    return Yup.number()
  })
}

export const instanceWarmupSchema = (getString: UseStringsReturn['getString']): Yup.Schema<number | undefined> => {
  return Yup.lazy((value: number): any => {
    /* istanbul ignore else */
    if (getMultiTypeFromValue(value as any) === MultiTypeInputType.FIXED) {
      return Yup.mixed().test({
        test(): boolean | Yup.ValidationError {
          if (!value) {
            return true
          }
          if (value < 0) {
            return this.createError({ message: getString('cd.instanceWarmupError') })
          }
          return true
        }
      })
    }
    return Yup.number()
  })
}
