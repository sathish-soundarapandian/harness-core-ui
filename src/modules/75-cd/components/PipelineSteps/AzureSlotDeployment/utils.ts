import { defaultTo } from 'lodash-es'
import { RUNTIME_INPUT_VALUE, MultiTypeInputType } from '@harness/uicore'
import type { StageElementWrapper } from '@pipeline/utils/pipelineTypes'
import type { StageElementConfig } from 'services/pipeline-ng'
import type { InfraStructureDefinitionYaml, DeploymentStageConfig, EnvironmentYamlV2 } from 'services/cd-ng'

import { getStageFromPipeline } from '@pipeline/components/PipelineStudio/StepUtil'

interface EnvironmentYamlV2Runtime extends Omit<EnvironmentYamlV2, 'infrastructureDefinitions'> {
  infrastructureDefinitions?: InfraStructureDefinitionYaml[] | string
}

interface DeploymentStageConfigAzureWebName extends Omit<DeploymentStageConfig, 'environment'> {
  environment: EnvironmentYamlV2Runtime
}

export interface DeploymentStageElementConfigAzureWebName extends Omit<StageElementConfig, 'spec'> {
  spec?: DeploymentStageConfigAzureWebName
}
export type SelectedStageType = StageElementWrapper<DeploymentStageElementConfigAzureWebName>

export const isRuntimeEnvId = (selectedStage: SelectedStageType): boolean => {
  const envId = selectedStage?.stage?.spec?.environment?.environmentRef
  return !!(envId && envId === RUNTIME_INPUT_VALUE)
}

export const isRuntimeInfraId = (selectedStage: SelectedStageType): boolean => {
  const infraId = selectedStage?.stage?.spec?.environment?.infrastructureDefinitions
  return infraId === RUNTIME_INPUT_VALUE
}

export const isMultiEnv = (selectedStage: SelectedStageType): boolean => {
  const isEnvs = !!selectedStage?.stage?.spec?.environments || !!selectedStage?.stage?.spec?.environmentGroup
  return !!isEnvs
}

export const getEnvId = (selectedStage: SelectedStageType): string => {
  if (isMultiEnv(selectedStage || isRuntimeEnvId(selectedStage))) {
    return ''
  }

  return defaultTo(selectedStage?.stage?.spec?.environment?.environmentRef, '')
}

export const getInfraId = (selectedStage: SelectedStageType): string => {
  const envId = getEnvId(selectedStage)
  if (!envId || isMultiEnv(selectedStage)) {
    return ''
  }

  if (!isRuntimeEnvId(selectedStage) && !isRuntimeInfraId(selectedStage)) {
    const [infra] = selectedStage?.stage?.spec?.environment?.infrastructureDefinitions as InfraStructureDefinitionYaml[]
    return infra.identifier
  }

  return ''
}

export const getEnvIdRuntime = (stageId: string, values: any) => {
  const stage = getStageFromPipeline(stageId, values) as SelectedStageType
  const envId = stage?.stage?.spec?.environment?.environmentRef
  return envId
}

export const getInfraIdRuntime = (stageId: string, values: any): string => {
  const stage = getStageFromPipeline(stageId, values) as SelectedStageType

  if (stage?.stage?.spec?.environment?.infrastructureDefinitions?.[0]) {
    const [{ identifier }] = stage?.stage?.spec?.environment
      ?.infrastructureDefinitions as InfraStructureDefinitionYaml[]
    return identifier
  }
  return ''
}

export const getAllowableTypes = (selectedStage: SelectedStageType): MultiTypeInputType[] => {
  if (isMultiEnv(selectedStage)) {
    return [MultiTypeInputType.EXPRESSION]
  }

  return [MultiTypeInputType.FIXED, MultiTypeInputType.RUNTIME, MultiTypeInputType.EXPRESSION]
}
