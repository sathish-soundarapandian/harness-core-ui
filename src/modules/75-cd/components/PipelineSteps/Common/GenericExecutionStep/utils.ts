/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { FormikErrors, yupToFormErrors } from 'formik'
import * as Yup from 'yup'
import { getMultiTypeFromValue, MultiTypeInputType } from '@harness/uicore'

import type { StepElementConfig } from 'services/cd-ng'
import { getDurationValidationSchema } from '@common/components/MultiTypeDuration/MultiTypeDuration'
import { ValidateInputSetProps, StepViewType } from '@pipeline/components/AbstractSteps/Step'

export const validateGenericFields = ({
  data,
  template,
  getString,
  viewType
}: ValidateInputSetProps<StepElementConfig>): FormikErrors<StepElementConfig> => {
  const isRequired = viewType === StepViewType.DeploymentForm || viewType === StepViewType.TriggerForm
  const errors: FormikErrors<StepElementConfig> = {}

  if (getMultiTypeFromValue(template?.timeout) === MultiTypeInputType.RUNTIME) {
    let timeoutSchema = getDurationValidationSchema({ minimum: '10s' })
    if (isRequired) {
      timeoutSchema = timeoutSchema.required(getString?.('validation.timeout10SecMinimum'))
    }
    const timeout = Yup.object().shape({
      timeout: timeoutSchema
    })
    try {
      timeout.validateSync(data)
    } catch (e) {
      if (e instanceof Yup.ValidationError) {
        const err = yupToFormErrors(e)
        Object.assign(errors, err)
      }
    }
  }
  return errors
}
