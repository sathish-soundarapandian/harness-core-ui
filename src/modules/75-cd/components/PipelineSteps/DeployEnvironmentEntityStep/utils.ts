/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import type { SelectOption } from '@harness/uicore'
import { defaultTo, isEmpty } from 'lodash-es'
import type { DeploymentStageConfig } from 'services/cd-ng'
import type { StringsMap } from 'stringTypes'

export interface DeployEnvironmentEntityCustomStepProps {
  getString: (key: keyof StringsMap) => string
  stageIdentifier: string
  serviceRef?: string
  environmentRef?: string
  infrastructureRef?: string
  clusterRef?: string
}
export interface DeployEnvironmentEntityConfig {
  environment?: DeploymentStageConfig['environment']
  environments?: DeploymentStageConfig['environments']
  gitOpsEnabled?: DeploymentStageConfig['gitOpsEnabled']
}

export interface DeployEnvironmentEntityFormState {
  environment?: DeploymentStageConfig['environment'] & {
    infrastructureRef?: string
  }
  environments?: Omit<DeploymentStageConfig['environments'], 'values'> & {
    values: SelectOption[]
  }
  gitOpsEnabled?: DeploymentStageConfig['gitOpsEnabled']
}

export function processInitialValues(initialValues: DeployEnvironmentEntityConfig): DeployEnvironmentEntityFormState {
  if (initialValues.environments) {
    return {
      environments: {
        values: defaultTo(initialValues.environments?.values, []).map(value => ({
          label: value.environmentRef,
          value: value.environmentRef
        }))
      }
    }
  } else {
    return {
      environment: {
        environmentRef: defaultTo(initialValues.environment?.environmentRef, ''),
        deployToAll: initialValues.environment?.deployToAll,
        infrastructureRef: defaultTo(initialValues.environment?.infrastructureDefinition?.identifier, '')
      }
    }
  }
}

export function processFormValues(
  data: DeployEnvironmentEntityFormState,
  initialValues: DeployEnvironmentEntityConfig
): DeployEnvironmentEntityConfig {
  if (!isEmpty(data.environments?.values)) {
    return {
      environments: {
        values: data.environments?.values.map(value => ({
          environmentRef: value.value as string
        }))
      }
    }
  } else if (data.environment) {
    if (data.environment?.environmentRef === '' || initialValues.environment?.environmentRef === '') {
      return {
        environment: {
          environmentRef: defaultTo(data.environment?.environmentRef, ''),
          deployToAll: false
        }
      }
    }

    return {
      // ...data,
      environment: {
        environmentRef: defaultTo(data.environment?.environmentRef, ''),
        deployToAll: false,
        infrastructureDefinition: {
          identifier: defaultTo(data.environment?.infrastructureRef, '')
        }
      }
    }
  } else {
    // TODO: Add default conversions
    return {}
  }
}
