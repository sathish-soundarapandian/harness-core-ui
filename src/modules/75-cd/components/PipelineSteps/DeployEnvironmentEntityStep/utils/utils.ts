/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { RUNTIME_INPUT_VALUE } from '@harness/uicore'
import { isEmpty, isEqual, isNil } from 'lodash-es'
import * as Yup from 'yup'
import type { UseStringsReturn } from 'framework/strings'
import { isValueRuntimeInput } from '@common/utils/utils'
import type {
  DeployEnvironmentEntityConfig,
  DeployEnvironmentEntityCustomStepProps,
  DeployEnvironmentEntityFormState
} from '../types'
import {
  processEnvironmentGroupInitialValues,
  processMultiEnvironmentInitialValues,
  processSingleEnvironmentGitOpsInitialValues,
  processSingleEnvironmentInitialValues
} from './processInitialValuesUtils'
import {
  processEnvironmentGroupFormValues,
  processMultiEnvironmentFormValues,
  processSingleEnvironmentFormValues,
  processSingleEnvironmentGitOpsFormValues
} from './processFormValuesUtils'

export function processInitialValues(
  initialValues: DeployEnvironmentEntityConfig,
  customStepProps: DeployEnvironmentEntityCustomStepProps,
  onUpdate?: (data: DeployEnvironmentEntityConfig) => void
): DeployEnvironmentEntityFormState {
  const gitOpsEnabled = !!customStepProps.gitOpsEnabled

  // istanbul ignore else
  if (initialValues.environment && initialValues.environment.environmentRef) {
    if (gitOpsEnabled) {
      return processSingleEnvironmentGitOpsInitialValues(initialValues.environment)
    } else {
      // this is backwards compatibility for single env multiple clusters in the older redesign setup
      if (
        Array.isArray(initialValues.environment.gitOpsClusters) &&
        initialValues.environment.gitOpsClusters.length > 1
      ) {
        return processMultiEnvironmentInitialValues(
          {
            environments: {
              metadata: {
                parallel: false
              },
              values: [{ ...initialValues.environment }]
            }
          },
          customStepProps
        )
      }
      return processSingleEnvironmentInitialValues(initialValues.environment, customStepProps)
    }
  } else if (initialValues.environments) {
    return processMultiEnvironmentInitialValues(initialValues, customStepProps)
  } else if (initialValues.environmentGroup) {
    // This handles the migration from runtime environment group of old pipeline to post multi infra handling
    if (
      isEqual(initialValues.environmentGroup, {
        envGroupRef: RUNTIME_INPUT_VALUE,
        deployToAll: true
      })
    ) {
      const updatedInitialValues = {
        environmentGroup: {
          envGroupRef: RUNTIME_INPUT_VALUE,
          deployToAll: RUNTIME_INPUT_VALUE as any,
          environments: RUNTIME_INPUT_VALUE as any
        }
      }

      onUpdate?.(updatedInitialValues)
      return processEnvironmentGroupInitialValues(updatedInitialValues, customStepProps)
    } else if (
      // This handles the migration from 'all' environments under environment group of old pipeline to post multi infra handling
      initialValues.environmentGroup.deployToAll === true &&
      !isValueRuntimeInput(initialValues.environmentGroup.envGroupRef) &&
      isNil(initialValues.environmentGroup.envGroupRef)
    ) {
      const updatedInitialValues = {
        environmentGroup: {
          envGroupRef: initialValues.environmentGroup.envGroupRef,
          deployToAll: true,
          environments: RUNTIME_INPUT_VALUE as any
        }
      }

      onUpdate?.(updatedInitialValues)
      return processEnvironmentGroupInitialValues(updatedInitialValues, customStepProps)
    }

    return processEnvironmentGroupInitialValues(initialValues, customStepProps)
  }

  return {
    category: 'single'
  }
}

export function processFormValues(
  data: DeployEnvironmentEntityFormState,
  customStepProps: DeployEnvironmentEntityCustomStepProps
): DeployEnvironmentEntityConfig {
  const gitOpsEnabled = !!customStepProps.gitOpsEnabled

  // istanbul ignore else
  if (data.category === 'single') {
    if (gitOpsEnabled) {
      return processSingleEnvironmentGitOpsFormValues(data)
    } else {
      return processSingleEnvironmentFormValues(data, customStepProps)
    }
  } else if (data.category === 'group') {
    return processEnvironmentGroupFormValues(data, customStepProps)
  } else if (data.category === 'multi') {
    return processMultiEnvironmentFormValues(data, customStepProps)
  }
  return {}
}

export function getValidationSchema(getString: UseStringsReturn['getString'], gitOpsEnabled: boolean): Yup.MixedSchema {
  return Yup.mixed()
    .required()
    .test({
      test(valueObj: DeployEnvironmentEntityFormState): boolean | Yup.ValidationError {
        if (valueObj.category === 'single') {
          if (!valueObj.environment) {
            return this.createError({
              path: 'environment',
              message: getString('cd.pipelineSteps.environmentTab.environmentIsRequired')
            })
          }

          if (!gitOpsEnabled && !valueObj.infrastructure && !isValueRuntimeInput(valueObj.environment)) {
            return this.createError({
              path: 'infrastructure',
              message: getString('cd.pipelineSteps.environmentTab.infrastructureIsRequired')
            })
          }

          // To be put back once BE supports multi environments
          // if (gitOpsEnabled && !valueObj.cluster && !isValueRuntimeInput(valueObj.environment)) {
          //   return this.createError({
          //     path: 'cluster',
          //     message: getString('cd.pipelineSteps.environmentTab.clusterIsRequired')
          //   })
          // }
        } else if (valueObj.category === 'multi') {
          if (
            !(Array.isArray(valueObj.environments) && valueObj.environments.length) &&
            !isValueRuntimeInput(valueObj.environments as unknown as string) &&
            isEmpty(valueObj.environmentFilters?.['fixedScenario'])
          ) {
            return this.createError({
              path: 'environments',
              message: getString('cd.pipelineSteps.environmentTab.environmentsAreRequired')
            })
          }
        } else if (valueObj.category === 'group') {
          if (!valueObj.environmentGroup) {
            return this.createError({
              path: 'environmentGroup',
              message: getString('cd.pipelineSteps.environmentTab.environmentGroupIsRequired')
            })
          }
        }

        return true
      }
    })
}
