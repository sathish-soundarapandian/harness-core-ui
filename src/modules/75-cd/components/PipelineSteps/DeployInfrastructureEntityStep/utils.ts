/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { defaultTo, isEmpty, isNil, merge } from 'lodash-es'

import type { InfraStructureDefinitionYaml } from 'services/cd-ng'

import { StepViewType } from '@pipeline/components/AbstractSteps/Step'
import { clearRuntimeInput } from '@pipeline/utils/runPipelineUtils'

import type { InfrastructureData } from '../DeployEnvironmentEntityStep/types'

export const createInfraTemplate = (
  existingTemplate: InfraStructureDefinitionYaml[],
  infrastructureIdentifiers: string[],
  infrastructuresData: InfrastructureData[]
): InfraStructureDefinitionYaml[] => {
  return infrastructureIdentifiers.map(infraIdentifier => {
    const existingTemplateOfInfrastructure = Array.isArray(existingTemplate)
      ? existingTemplate.find(infraTemplate => (infraTemplate as any).infraIdForTemplate === infraIdentifier)
      : undefined

    const templateFromInfrastructureData = infrastructuresData.find(
      infraTemplate => infraTemplate.infrastructureDefinition.identifier === infraIdentifier
    )

    const newInputsTemplate = defaultTo(
      existingTemplateOfInfrastructure?.inputs,
      templateFromInfrastructureData?.infrastructureInputs
    )

    return {
      identifier: infraIdentifier,
      inputs: newInputsTemplate
    }
  })
}

export const createInfraValues = (
  existingInfrastructureValues: InfraStructureDefinitionYaml[],
  infrastructureIdentifiers: string[],
  infrastructuresData: InfrastructureData[],
  deployToAllInfrastructures: boolean,
  stepViewType?: StepViewType
): InfraStructureDefinitionYaml[] => {
  return infrastructureIdentifiers.map(infraId => {
    const valueFromInfrastructuresData = infrastructuresData.find(
      infraTemplate => infraTemplate.infrastructureDefinition.identifier === infraId
    )?.infrastructureInputs

    // Start - Retain form values

    const existingValueOfInfrastructure = Array.isArray(existingInfrastructureValues)
      ? existingInfrastructureValues.find(infraValue => infraValue.identifier === infraId)
      : undefined

    const newInfrastructureObject = !isNil(existingValueOfInfrastructure)
      ? existingValueOfInfrastructure
      : ({} as InfraStructureDefinitionYaml)

    let infrastructureInputs = newInfrastructureObject.inputs
    const infrastructureInputsFromInfrastructureData = valueFromInfrastructuresData?.inputs

    const baseInfrastructureInputs = !isEmpty(infrastructureInputsFromInfrastructureData)
      ? deployToAllInfrastructures && stepViewType === StepViewType.TemplateUsage
        ? infrastructureInputsFromInfrastructureData
        : clearRuntimeInput(infrastructureInputsFromInfrastructureData)
      : undefined

    if (!infrastructureInputs) {
      infrastructureInputs = baseInfrastructureInputs
    } else {
      infrastructureInputs = merge(baseInfrastructureInputs, infrastructureInputs)
    }

    // End - Retain form values

    return {
      identifier: infraId,
      inputs: infrastructureInputs
    }
  })
}
