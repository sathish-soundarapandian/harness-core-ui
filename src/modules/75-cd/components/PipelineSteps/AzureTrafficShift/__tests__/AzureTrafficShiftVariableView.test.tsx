/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { render } from '@testing-library/react'
import { StepType } from '@pipeline/components/PipelineSteps/PipelineStepInterface'

import { TestWrapper } from '@common/utils/testUtils'
import type { AzureTrafficShiftVariableStepProps } from '../AzureTrafficShiftInterface.types'
import { AzureTrafficShiftVariableStep } from '../AzureTrafficShiftVariableView'
jest.mock('@common/components/YAMLBuilder/YamlBuilder')

const props = {
  initialValues: {
    type: 'AzureTrafficShift',
    name: 'ats',
    identifier: 'ats',
    timeout: '10s',
    spec: {
      traffic: '20%'
    }
  },
  originalData: {
    type: 'AzureTrafficShift',
    name: 'ats',
    identifier: 'ats',
    timeout: '10s',
    spec: {
      traffic: '20%'
    }
  },
  stageIdentifier: 'qaStage',
  onUpdate: jest.fn(),
  metadataMap: {
    'step-name': {
      yamlProperties: {
        fqn: 'pipeline.stages.qaStage.execution.steps.ats.name',
        localName: 'step.ats.name'
      }
    },
    'step-timeout': {
      yamlProperties: {
        fqn: 'pipeline.stages.qaStage.execution.steps.ats.timeout',
        localName: 'step.ats.timeout'
      }
    },
    'step-traffic': {
      yamlProperties: {
        fqn: 'pipeline.stages.qaStage.execution.steps.ats.spec.traffic',
        localName: 'step.ats.spec.traffic'
      }
    }
  },
  stepType: StepType.AzureTrafficShift
} as AzureTrafficShiftVariableStepProps

describe('Azure Traffic Shift Variable view ', () => {
  test('Initial render', () => {
    const { container } = render(
      <AzureTrafficShiftVariableStep
        initialValues={{
          type: 'AzureTrafficShift',
          name: 'ats',
          identifier: 'ats',
          timeout: '10m',
          spec: {
            traffic: '20%'
          }
        }}
        stepType={StepType.AzureTrafficShift}
        onUpdate={() => jest.fn()}
        {...{
          stageIdentifier: 'qaStage',
          metadataMap: props.metadataMap,
          variablesData: props.variablesData
        }}
      />
    )

    expect(container).toMatchSnapshot()
  })

  test('Initial render inline with no values', () => {
    const { container } = render(
      <TestWrapper>
        <AzureTrafficShiftVariableStep
          initialValues={{
            type: 'AzureTrafficShift',
            name: 'ats',
            identifier: 'ats',
            timeout: '10m',
            spec: {
              traffic: '20%'
            }
          }}
          stepType={StepType.AzureTrafficShift}
          onUpdate={() => jest.fn()}
          {...{
            stageIdentifier: 'qaStage',
            metadataMap: props.metadataMap,
            variablesData: props.variablesData
          }}
        />
      </TestWrapper>
    )

    expect(container).toMatchSnapshot()
  })

  test('Should render with inline config', () => {
    const { container } = render(
      <TestWrapper>
        <AzureTrafficShiftVariableStep {...props} />
      </TestWrapper>
    )
    expect(container).toMatchSnapshot()
  })

  test('should render with no config', () => {
    const { container } = render(
      <TestWrapper>
        <AzureTrafficShiftVariableStep
          {...props}
          initialValues={{
            type: 'AzureTrafficShift',
            name: 'ats',
            identifier: 'ats',
            timeout: '10m',
            spec: {
              traffic: '20%'
            }
          }}
        />
      </TestWrapper>
    )
    expect(container).toMatchSnapshot()
  })
})
