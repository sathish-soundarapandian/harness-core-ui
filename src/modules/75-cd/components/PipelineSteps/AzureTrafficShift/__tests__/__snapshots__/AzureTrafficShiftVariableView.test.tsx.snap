// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Azure Traffic Shift Variable view  Initial render 1`] = `
<div>
  <div
    class="variablesListTable variablePaddingL3"
  />
</div>
`;

exports[`Azure Traffic Shift Variable view  Initial render inline with no values 1`] = `
<div>
  <div
    class="variablesListTable variablePaddingL3"
  />
</div>
`;

exports[`Azure Traffic Shift Variable view  Should render with inline config 1`] = `
<div>
  <div
    class="variablesListTable variablePaddingL3"
  />
</div>
`;

exports[`Azure Traffic Shift Variable view  should render with no config 1`] = `
<div>
  <div
    class="variablesListTable variablePaddingL3"
  />
</div>
`;
